projects/common/src/telemetry/user-telemetry-internal.service.ts

```diff
@@ -0,0 +1,144 @@
+import { Injectable, Injector } from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+import { Observable, ReplaySubject, Subject } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { Dictionary } from '../utilities/types/types';
+import { UserTelemetryProvider, UserTelemetryRegistrationConfig, UserTraits } from './telemetry';
+
+@Injectable({ providedIn: 'root' })
+export class UserTelemetryInternalService {
+  private telemetryProviders: UserTelemetryInternalConfig[] = [];
+  private readonly telemetryActionSubject: Subject<TelemetryAction> = new ReplaySubject();
+  private readonly telemetryAction$: Observable<TelemetryAction>;
+  private readonly identifyAction$: Observable<TelemetryAction>;
+  private readonly trackEventAction$: Observable<TelemetryAction>;
+  private readonly trackPageAction$: Observable<TelemetryAction>;
+  private readonly trackErrorAction$: Observable<TelemetryAction>;
+
+  public constructor(private readonly injector: Injector, private readonly router: Router) {
+    this.telemetryAction$ = this.telemetryActionSubject.asObservable();
+
+    this.identifyAction$ = this.telemetryAction$.pipe(filter(action => action.type === TelemetryActionType.Identify));
+
+    this.trackEventAction$ = this.telemetryAction$.pipe(
+      filter(action => action.type === TelemetryActionType.TrackEvent)
+    );
+
+    this.trackPageAction$ = this.telemetryAction$.pipe(
+      filter(action => action.type === TelemetryActionType.TrackPageView)
+    );
+
+    this.trackErrorAction$ = this.telemetryAction$.pipe(
+      filter(action => action.type === TelemetryActionType.TrackError)
+    );
+
+    this.setupAutomaticPageTracking();
+    this.setupTelemetryActionHandlers();
+  }
+
+  public register(...configs: UserTelemetryRegistrationConfig<unknown>[]): void {
+    try {
+      const providers = configs.map(config => this.buildTelemetryProvider(config));
+      this.telemetryProviders = [...this.telemetryProviders, ...providers];
+    } catch (error) {
+      /**
+       * NoOp
+       */
+    }
+  }
+
+  public identify(userTraits: UserTraits): void {
+    this.telemetryActionSubject.next({ type: TelemetryActionType.Identify, data: userTraits });
+  }
+
+  public shutdown(): void {
```

Comment:
do we have any requirements that shutdown be called? who's in charge of calling it?

Reply:
The main service in-app (which is also calling the initialize method)

Reply:
Since we would only want to capture data on prod env, a separate initialize method would be useful

Reply:
my bad. I removed the `initialize`call in the latest code. I need to bring it back.

Reply:
How does that know when the app is shutting down? I can't remember, do we get an onDestroy hook for the root component or something?

Reply:
Yeah, the application frame component can call it on destroy. Or we can call it when the corresponding feature flag is turned off after the the UI is initialized. 

I don't have a strong use case for it. Maybe I can go and ready why these telemetry platforms expose this method in the first place.